File: baseapp/src/main/java/ua/com/todd/baseapp/fragment/FragmentFactory.java
package ua.com.todd.baseapp.fragment;

import android.app.Fragment;
import android.os.Bundle;

public abstract class FragmentFactory {
    public Fragment getFragment(IBaseFragmentType type){
        return getFragment(type, null);
    }

    public abstract Fragment getFragment(IBaseFragmentType type, Bundle bundle);

    public interface IBaseFragmentType {}
}

File: app/src/main/kotlin/ua/com/todd/smsforwading/fragment/FragmentFactory.kt
package ua.com.todd.smsforwading.fragment

import ua.com.todd.baseapp.ui.fragment.FragmentFactory.IBaseFragmentType
import android.os.Bundle
import android.app.Fragment
import ua.com.todd.baseapp.ui.fragment

public class FragmentFactory : fragment.FragmentFactory() {
    override fun getFragment(type: IBaseFragmentType, bundle: Bundle?) = when (type as FragmentType) {
        FragmentType.SETTINGS -> SettingsFragment.getInstance()
        FragmentType.PROFILE -> ProfileFragment.getInstance()
        FragmentType.ITEM -> AddItemFragment.getInstance()
    }

    public enum class FragmentType : IBaseFragmentType {
        SETTINGS
        PROFILE
        ITEM
    }
}
